library/ssl_tls.c

```diff
@@ -2838,17 +2853,25 @@ static int ssl_session_save( const mbedtls_ssl_session *session,
     }
 
     /* Forward to version-specific serialization routine. */
+    remaining_len = used <= buf_len ? buf_len - used : 0;
```

Comment:
(very minor) if you need to make other changes, this might be more readable as

`remaining_len = (buf_len >= used) ? buf_len - used : 0;`

1) having parentheses round the condition puts the reader in the mindset of an `if` statement;
2) having the two variables in the same order makes it clearer.

However, don't bother if there's nothing else to change - this is such a minor thing

Reply:
I like your suggestion.